frame/contracts/rpc/src/lib.rs

```diff
@@ -237,32 +243,31 @@ where
 }
 
 /// Converts a runtime trap into an RPC error.
-fn runtime_error_into_rpc_err(err: impl std::fmt::Debug) -> Error {
-	Error {
-		code: ErrorCode::ServerError(RUNTIME_ERROR),
+fn runtime_error_into_rpc_err(err: impl std::fmt::Debug) -> JsonRpseeError {
+	CallError::Custom {
+		code: RUNTIME_ERROR,
 		message: "Runtime error".into(),
-		data: Some(format!("{:?}", err).into()),
+		data: to_raw_value(&format!("{:?}", err)).ok(),
 	}
+	.into()
 }
 
-fn decode_hex<H: std::fmt::Debug + Copy, T: TryFrom<H>>(from: H, name: &str) -> Result<T> {
-	from.try_into().map_err(|_| Error {
-		code: ErrorCode::InvalidParams,
-		message: format!("{:?} does not fit into the {} type", from, name),
-		data: None,
-	})
+fn decode_hex<H: std::fmt::Debug + Copy, T: TryFrom<H>>(
+	from: H,
+	name: &str,
+) -> Result<T, JsonRpseeError> {
+	from.try_into()
+		.map_err(|_| anyhow!("{:?} does not fit into the {} type", from, name).into())
 }
 
-fn limit_gas(gas_limit: Weight) -> Result<()> {
+fn limit_gas(gas_limit: Weight) -> Result<(), JsonRpseeError> {
 	if gas_limit > GAS_LIMIT {
-		Err(Error {
-			code: ErrorCode::InvalidParams,
-			message: format!(
-				"Requested gas limit is greater than maximum allowed: {} > {}",
-				gas_limit, GAS_LIMIT
-			),
-			data: None,
-		})
+		Err(anyhow!(
+			"Requested gas limit is greater than maximum allowed: {} > {}",
+			gas_limit,
+			GAS_LIMIT
+		)
+		.into())
```

Comment:
Shouldn't this be typed error instead of `anyhow`?

Reply:
Maaaybe. I mean, typed errors are always good ofc but here we're returning an [`ErrorObject`](https://github.com/paritytech/jsonrpsee/blob/aacf7c0ecdb71da345e7c5cb0283f5cb5a040bd7/types/src/v2/error.rs#L56) ([this](https://docs.rs/jsonrpc-core/18.0.0/jsonrpc_core/types/error/struct.Error.html) in `jsonrpc-core`) so it'd be thrown away immediately anyway.
[Original code](https://github.com/paritytech/substrate/blob/5d22361ebbd9b5ccafd87f3f50d2646ad59b6c0b/frame/contracts/rpc/src/lib.rs#L258).

Reply:
I would no because this error just care about the error message in `JSON-RPC error` and no need to set a specific error code and so on IMHO.